### PR TITLE
Avoid non-transactional modification of relfrozenxid during CLUSTER

### LINKED ISSUE #10150: CLUSTER command fails when it's run after previously aborted CLUSTER.
### Greenplum version or build
master (7x)
### Expected behavior
CLUSTER command finishes successfully when it runs after a previously aborted CLUSTER command on the same table.

### Actual behavior
The second CLUSTER command fails.

### Step to reproduce the behavior
```SQL
create table foo (a int, b varchar, c int) distributed by (a);
insert into foo select i, 'initial insert' || i, i from generate_series(1,10000)i;
create index ifoo on foo using btree (b);
-- execute cluster in a transaction but don't commit the transaction
begin;
cluster foo using ifoo; 
abort;

-- try cluster again, this time it fails
cluster foo using ifoo;
ERROR:  found xmin 622 from before relfrozenxid 623  (seg2 127.0.0.1:7004 pid=8696)
```

### PR DESCRIPTION
**Pull request by x4m**
Currently function swap_relation_files() calls non-transactional vac_update_relstats() in transactional context during CLUSTER.
This change prevents emitting corrupted frozen xid and min multixact. I think problem was introduced in #7040.
I've tried to produce minimal sane diff to fix #10150. Probably, we could avoid calling vac_update_relstats() at all, but it sends useful message to QD.

### CODE REVIEW

File: src/test/regress/sql/cluster.sql [hunk 1]
```diff
@@ -290,3 +296,15 @@ DROP TABLE clstr_2;
...
+CREATE TABLE foo (a int, b varchar, c int) DISTRIBUTED BY (a);
+INSERT INTO foo SELECT i, 'initial insert' || i, i FROM generate_series(1,10000)i;
+CREATE index ifoo on foo using btree (b);
+-- execute cluster in a transaction but don't commit the transaction
+BEGIN;
+CLUSTER foo USING ifoo;
+ABORT;
+-- try cluster again
+CLUSTER foo USING ifoo;
+DROP TABLE foo;
```
Comment (ashwinstar) [inline src/test/regress/sql/cluster.sql:310]: Please add new line here.

Comment (x4m) [reply src/test/regress/sql/cluster.sql]: done

**Comment by ashwinstar** [comment]
Looks good to me. Thank You. I am commit the same.

**Comment by ashwinstar** [comment]
Pushed this via commit 7f7fa49820112ba8d9d0ad688b84816b575c4560. Thanks @x4m for contribution.

**Comment by x4m** [comment]
Cool, thanks!

**Pull request closed without merging**